pkg/pulumiyaml/ast/template.go

```diff
@@ -102,16 +102,27 @@ func (d *ConfigMapDecl) parse(name string, node syntax.Node) syntax.Diagnostics
 	entries := make([]ConfigMapEntry, obj.Len(), obj.Len())
 	for i := range entries {
 		kvp := obj.Index(i)
-
-		var v *ConfigParamDecl
-		vname := fmt.Sprintf("%s.%s", name, kvp.Key.Value())
-		vdiags := parseField(vname, reflect.ValueOf(&v).Elem(), kvp.Value)
-		diags.Extend(vdiags...)
-
-		entries[i] = ConfigMapEntry{
-			syntax: kvp,
-			Key:    StringSyntax(kvp.Key),
-			Value:  v,
+		if _, ok := kvp.Value.(*syntax.ObjectNode); !ok {
```

Comment:
@AaronFriel `pulumi convert` was already calling `RewriteShorthandConfigValues()`, turning something like `config: aws:region: us-west-1` into `aws:region: value: us-west-1`; however, the node was still interpreted as its direct value/ string node here, not an object node. When we deprecate `configuration` I think we'd loosen the constraint here anyways